### ISSUE TITLE
Sprime help

### ISSUE THREAD

**Issue by wangzeng2001** [Yamei-Ding/Juglans#1]
Dear Ding,
I am a first-year master student in Sichuan university, I read your paper and impressed by your good job. You used a variety of ways to identify introgression, and i am trying to use the pipeline of SPrime. However, my result makes me confused that half of the genome region is regarded as introgression region. I don't quite understand the filter step in the paper  "Analysis of Human Sequence Data Reveals Two Pulses of Archaic Denisovan Admixture", and my research object is also plants, so I would like to ask about your filter method. I have tried hard and contacted the author of SPrime, but there is still no solution. So i turned to your help!  I attached my step in Chinese below 
[my step.docx](https://github.com/Yamei-Ding/Juglans/files/10089899/my.step.docx)
. Thanks a million!!!

**Comment by Yamei-Ding** [comment]
I see your pipeline for running Sprime. There are two things you might need to check. First, the input map file for Sprime needs genetic distance for each site with cM units. Second, as far as I know, the Sprime is usually used to detect the archaic/ghost introgression for target population. The premise for Sprime method is that the outgroup population  is closely related to the target population but that is not experienced introgression from archaic population (Browning et al. 2018).

**Comment by wangzeng2001** [comment]
Thanks a lot for your reply, which is very helpful to me. So the third column of the map file needs to calculate the genetic distance of two SNPs adjacent to each other, is it right？Take the liberty of asking，is it very time-consuming to calculate this with LDhat for Millions of locus in your paper?

**Comment by Yamei-Ding** [comment]
Yes, the running time is related to the sample size and the number of SNPs. It took us about two weeks.

**Comment by wangzeng2001** [comment]
Sorry to bother you again. I used LDhat software to calculate the recombination rate of high-quality sites, and then calculated the genetic distance using the formula in your paper. If the result of the calculation is used directly as the third column, the data in the third column is not in ascending order, but the software requires that it be ascending. I see that the test file of the sprime software has a third column of 0 at the first locus of chromosomes and a 1 at the last site. According to my understanding, this column counts the genetic distance of each site from the beginning site of chromosomes, and I am still very confused and do not understand how to calculate it. After calculating the recombination rate and genetic distance for each site, how do you calculate your third column?
